metagpt/roles/di/data_interpreter.py

```diff
@@ -51,14 +99,30 @@ async def _write_and_exec_code(self, max_retry: int = 3):
         counter = 0
         success = False
 
+        # plan info
+        plan_status = self.planner.get_plan_status() if self.use_plan else ""
+
+        # tool info
+        if self.tools:
+            context = (
+                self.working_memory.get()[-1].content if self.working_memory.get() else ""
+            )  # thoughts from _think stage in 'react' mode
+            plan = self.planner.plan if self.use_plan else None
+            tool_info = await self.tool_recommender.get_recommended_tool_info(context=context, plan=plan)
+        else:
+            tool_info = ""
+
+        # data info
+        await self._check_data()
```

Comment:
I still recommend writing similar functions in a reentrant form rather than with state.

Reply:
Since _check_data is not meant to be used independently or externally, but only encapsulated to hide details and make this step clearer in main pipeline, would passing args be a bit unnecessary?